src/components/Explorer/index.tsx

```diff
@@ -1,48 +1,86 @@
+'use client'
+
+import _ from 'lodash'
 import {
-  Code2,
   Cog,
   Cpu,
   FileJson,
-  Film,
   Joystick,
   MoreHorizontal,
   Terminal,
-} from 'lucide-react';
-import { OpenFilesSubMenu } from '../OpenFilesTabs/OpenFilesSubMenu';
-import { File } from './File';
-import { Folder } from './Folder';
-import { SubMenu } from './SubMenu';
+  type LucideIcon,
+} from 'lucide-react'
+import { OpenFilesSubMenu } from '../OpenFilesTabs/OpenFilesSubMenu'
+import { FolderStructure } from './FolderStructure'
+import { SubMenu } from './SubMenu'
 
 export type FileType = {
-  title: string;
-  icon: React.ReactNode;
-};
+  folder: string
+  title: string
+  icon: LucideIcon
+}
 
 export const explorerFiles: Record<string, FileType> = {
-  "/vscode/settings": {
-    icon: <FileJson size={16} />,
-    title: "settings.json",
+  '/vscode/settings': {
+    folder: 'Visual Studio Code',
+    title: 'settings.json',
+    icon: FileJson,
+  },
+  '/vscode/extensions': {
+    folder: 'Visual Studio Code',
+    title: 'extensions.json',
+    icon: FileJson,
   },
-  "/vscode/extensions": {
-    icon: <FileJson size={16} />,
-    title: "extensions.json",
+  '/terminal/general': {
+    folder: 'Terminal',
+    title: 'General',
+    icon: Terminal,
   },
-  "/terminal/general": {
-    icon: <Terminal size={16} />,
-    title: "General",
+  '/terminal/fish': {
+    folder: 'Terminal',
+    title: 'config.fish',
+    icon: Cog,
   },
-  "/terminal/fish": { icon: <Cog size={16} />, title: "config.fish" },
-  "/others/dev-setup": { icon: <Cpu size={16} />, title: "dev.setup" },
-  "/others/gaming-setup": {
-    icon: <Joystick size={16} />,
-    title: "gaming.setup",
+  '/others/dev-setup': {
+    folder: 'Others',
+    title: 'dev.setup',
+    icon: Cpu,
   },
-};
+  '/others/gaming-setup': {
+    folder: 'Others',
+    title: 'gaming.setup',
+    icon: Joystick,
+  },
+}
+
+// NOTE: We're using it to pass the file path into files array
+const newFolderStructure = Object.entries(explorerFiles).map(([path, file]) => {
+  return {
+    title: file.folder,
+    files: [
+      {
+        path,
+        name: file.title,
+        icon: file.icon,
+      },
+    ],
+  }
+})
```

Comment:
Can't we update the original array instead of creating a new one? I mean, you can change the format of the `explorerFiles` array so we won't need this.